drivers/Aqara/aqara-feeder/src/init.lua

```diff
@@ -104,23 +106,25 @@ local function petFeeder_handler(driver, device, value, zb_rx)
   elseif funcID == "14.92.85" then
     -- feed portion
     device:emit_event(capabilities.feederPortion.feedPortion({ value = conv_data(param), unit = "servings" }))
-  elseif funcID == "13.104.85" then
-    local feed_source = device:get_field(FEED_SOURCE)
+  elseif funcID == "13.104.85" and conv_data(param) ~= 0 then
+    local feed_source = device:get_field(FEED_SOURCE) or 0
     if feed_source == 0 then
       device:emit_event(capabilities.feederOperatingState.feederOperatingState("feeding"))
     end
     device:set_field(FEED_SOURCE, 0, { persist = true })
     delete_timer(device)
-    device:set_field(FEED_TIMER, device.thread:call_with_delay(FEED_TIME, callback_timer(driver, device)))
+    device:set_field(FEED_TIMER, device.thread:call_with_delay(FEED_TIME, callback_feed(device)))
+  elseif funcID == "13.11.85" then
+    -- error
+    delete_timer(device)
+    local evt = "idle"
+    if conv_data(param) == 1 then evt = "error" end
+    device:emit_event(capabilities.feederOperatingState.feederOperatingState(evt))
   end
 end
 
 -- [[ lifecycle_handlers ]]
 local function device_added(driver, device)
-  -- private protocol enable
-  device:send(cluster_base.write_manufacturer_specific_attribute(device,
-    PRIVATE_CLUSTER_ID, 0x0009, MFG_CODE, data_types.Uint8, 1))
-  -- init
   do_payload(device, 4, 24, 85, OP_WRITE, 1, 0)
```

Comment:
does this private protocol need to be enabled before the `do_payload` here will actually do anything?